src/components/ClientForm.vue

```diff
@@ -220,6 +220,43 @@
         Edit
       </button>
     </div>
+    <h2>Unique Attributes</h2>
```

Comment:
Maybe move this part to another `<section>`.

Also, would you like it to show up if the user is creating a new client?

Reply:
I'll probably move this into its own component and then add this to the add client page

Reply:
If you decide to keep it on the "New Client" page, you need to realize one thing: the `vm.client` used by the `newAtt` is unsaved. Therefore, at the moment the `Add Unique Key` button is pressed, the app will be saving a new `ClientAttributes` object whose `client` points to a `Client` object that doesn't exist in the database, which will lead to an error.

An idea I just came up with is to create a local "staging zone" for the unique attributes that are pending to be saved for the new client. These unique attributes only get saved once the "Save" button is hit for the new client. However, this logic would only work well for the "New Client" page and could be redundant for existing clients. There are probably better methods to use the minimum amount of code to reach the best user experience. Take some time to think how you can solve this problem gracefully.

Reply:
Another idea I have is to have custom attributes mingled with ordinary attributes. They can be created, edited, or deleted once the user enters the editing mode, but all the edits will not be saved until the user hits the "Save" button, which saves both ordinary attributes and custom attributes.

<img width="1069" alt="截屏2020-04-10 下午7 05 33" src="https://user-images.githubusercontent.com/44103628/79032824-43089680-7b5e-11ea-9ede-c6673b1a8258.png">

For example to be like this. The "New Field" will always appear after all other fields. It looks the same as other fields, but clicking on it will create a new attribute. There could be a small "Edit" icon next to the field name, and clicking on it allows the user to edit the label. There will also be a "Delete" icon for deleting the field.